griptape/config/base_structure_config.py

```diff
@@ -36,6 +38,26 @@ class BaseStructureConfig(BaseConfig, ABC):
     text_to_speech_driver: BaseTextToSpeechDriver = field(kw_only=True, metadata={"serializable": True})
     audio_transcription_driver: BaseAudioTranscriptionDriver = field(kw_only=True, metadata={"serializable": True})
 
+    _structure: Optional[Structure] = field(default=None, kw_only=True, alias="structure")
+
+    @property
+    def structure(self) -> Optional[Structure]:
+        return self._structure
+
+    @structure.setter
+    def structure(self, structure: Structure) -> None:
+        self._structure = structure
+
+        event_listener = EventListener(self.structure.publish_event)
+
+        self.prompt_driver.add_event_listener(event_listener)
+        self.image_generation_driver.add_event_listener(event_listener)
+        self.image_query_driver.add_event_listener(event_listener)
+        self.embedding_driver.add_event_listener(event_listener)
+        self.vector_store_driver.add_event_listener(event_listener)
+        if self.conversation_memory_driver is not None:
+            self.conversation_memory_driver.add_event_listener(event_listener)
+
     def merge_config(self, config: dict) -> BaseStructureConfig:
```

Comment:
What will happen if you call `merge_config` when `_structure` is set? Would it basically be the same behavior as before when the drivers had references to the structure?

Reply:
I'm not really sure. `merge_config` has all sorts of [issues](https://github.com/griptape-ai/griptape/issues/826) and should probably be removed.